cmd/ssl-watch/types.go

```diff
@@ -31,17 +32,70 @@ type Metrics struct {
 	mutex sync.RWMutex
 }
 
+type Domains struct {
+	db    map[string][]string
+	mutex sync.RWMutex
+}
+
 type App struct {
-	Domains map[string][]string `json:"domains"`
+	domains Domains
 	config  Config
 	log     jsonlog.Logger
 	metrics Metrics
 }
 
-func (m *Metrics) ListDomains() []string {
+func (d *Domains) Flush() {
+
+	d.mutex.Lock()
+	d.db = map[string][]string{}
+	d.mutex.Unlock()
+
+}
+
+func (d *Domains) Update(rawJSON []byte) {
+
+	d.mutex.Lock()
+	json.Unmarshal([]byte(rawJSON), &d.db)
+	d.mutex.Unlock()
 
+}
+
+func (d *Domains) List() []string {
+
+	defer d.mutex.RUnlock()
 	domains := []string{}
+	d.mutex.RLock()
+	for domain, _ := range d.db {
+		if strings.Contains(domain, ".") {
+			domains = append(domains, domain)
+		}
+	}
+	return domains
+}
+
+func (d *Domains) GetIPs(domain string) []string {
+
+	defer d.mutex.RUnlock()
+	ips := []string{}
+	d.mutex.RLock()
+
+	addrSet, ok := d.db[domain]
+	if ok && len(addrSet) > 0 {
+		if !strings.Contains(addrSet[0], ".") {
+			addrSet, ok = d.db[addrSet[0]]
+		}
+		for _, ip := range addrSet {
```

Comment:
having:
```
for i := range addrSet {
    ips = append(ips, addrSet[i])
}
```

Will make it one allocation less for each element in the loop. Not like this is valid point here, just for your information.